include/aikido/control/TrajectoryExecutor.hpp

```diff
@@ -11,14 +15,51 @@ namespace control {
 class TrajectoryExecutor
 {
 public:
-  virtual ~TrajectoryExecutor() = default;
+  virtual ~TrajectoryExecutor();
 
   /// Execute traj and set future upon completion.
   /// \param traj Trajectory to be executed.
-  virtual std::future<void> execute(trajectory::TrajectoryPtr _traj) = 0;
+  /// \param skip Whether to skip to the end of the trajectory (only used by
+  ///        simulation trajectory executors)
+  virtual std::future<void> execute(
+      trajectory::TrajectoryPtr _traj, bool _skip = false)
+      = 0;
 
   /// Executes one step.
   virtual void step() = 0;
+
+  /// Queue a trajectory for future execution. If there are no trajectories on
+  /// the queue or currently being executed, the trajectory should be executed
+  /// immediately.
+  /// \param traj Trajectory to queue and execute
+  void queue(trajectory::TrajectoryPtr _traj);
+
+  /// Toggle whether trajectories should be automatically dequeued and executed.
+  /// \param flag Whether to automatically execute trajectories from the queue
+  void setExecuteFromQueue(bool flag);
+
+  /// Block until all trajectories in the queue have been executed.
+  void executeAllFromQueue();
```

Comment:
Why is this function necessary? If we can't eliminate it entirely - which I think we should try to do - I suggest turning this into a non-blocking function that returns an `std::future<void>` that completes.